de.fu_berlin.inf.dpp/plugin.xml

```diff
@@ -569,7 +569,7 @@
       <handler
       class="de.fu_berlin.inf.dpp.ui.commandHandlers.ShareProjectHandler"
             commandId="de.fu_berlin.inf.dpp.ui.commands.ShareResource">
-         <activeWhen>
+         <enabledWhen>
```

Comment:
I had to use enabledWhen instead of activeWhen, because activeWhen leads to the behaviour that the "Share Project(s)..." entry is not enabled when the xmpp connection is started. It was always necessary to click into another view in order to enable the menu entry.
I don't understand 100% why this fix works and what was the problem with activeWhen in eclipse 4.6. The build result using #271 does not have this issue. Therefore I assume that eclipse 4.6 is the reason.

Reply:
So to my understanding the following applies.

You define a command and give it a name. (This will will be shown). Then you add this command to a section somewhere in Eclipse (context menu, menu etc.) There you can relabel it if you want

e.g
<command commandId="de.fu_berlin.inf.dpp.ui.commands.ShareResource" label="Multiple Contacts..." style="push">

And you can also choose if it should be visible or not.

Now the interesting part. A command can have multiple handlers. The activeWhen condition should evaluate if can handle such a command. Last but not least enabledWhen determines if the handler can execute the command.

See https://www.eclipsezone.com/eclipse/forums/t108159.html

Eclipse commands can have more than one handler. activeWhen is used by
the HandlerAuthority to 1) get the list of active handlers for a command
at a specific point in time and 2) pick a single active handler if there
is more than one active.

enabledWhen is simply declarative enablement for a given handler
(isEnabled(), true or false).


menu contributions visibleWhen determines visibility. But a command
with no active handler or a command with an active handler-not enabled
both simply show up as disabled in the menu/toolbar


> Just I was wondering is there any valid scenario where handler is active
> but not enabled. (activeWhen clause is satified and enableWhen clause is
> not satified)?

All the time ... for example, copy in the markers view is active when
the markers view is active, but might be disabled if you select a
category (which cannot be copied).

Reply:
This architecture is very confusing. I think I have superficial understanding of this handling. I think my change should not lead to an undesired behaviour in other parts of the GUI. Do you agree?

Reply:
But I am wondering why this switch works. Can you tell me what context menu entry in which view does not work as intended ?

Reply:
The menu `"Saros" > "Share Project(s)..."` in the main eclipse menu bar does not work as intended. In order to reproduce the issue in eclipse 4.6 (without this adjustment of the `plugin.xml`) you have to:
* Click the "Connect" button in the Saros view (I assume that an account is already configured)
* Open the menu `Saros` and you will see that `Share Project(s)...` is still grey and cannot be clicked (as before without a connected Saros).
* If you click into another view as e.g. the `Package Explorer` the menu entry `Share Project(s)...` becomes clickable.

I think that the click into another view retriggered the evaluation of properties that are used to determine the state of the `Share Project(s)...` entry.